### DISCUSSION TITLE
v0.24.0

### DISCUSSION THREAD

**Discussion by WithoutPants** [stashapp/stash#4398] [Announcements]
# Release Notes
### ✨ New Features
* Added manager for installing, updating and uninstalling scrapers and plugins. ([#4242](https://github.com/stashapp/stash/pull/4242))
* Added support for enabling and disabling plugins in the UI. ([#4141](https://github.com/stashapp/stash/pull/4141))
* Added support for plugin settings. ([#4143](https://github.com/stashapp/stash/pull/4143))
* Added support for plugin assets, external scripts and CSP overrides. ([#4260](https://github.com/stashapp/stash/pull/4260))
* Added UI plugin API to better support UI-based plugins. ([#4256](https://github.com/stashapp/stash/pull/4256))
* Added Studio Code and Photographer to Galleries. ([#4195](https://github.com/stashapp/stash/pull/4195))
* Added Details, Studio Code and Photographer to Images. ([#4217](https://github.com/stashapp/stash/pull/4217))
* Added scene framerate filter. ([#4161](https://github.com/stashapp/stash/pull/4161))
* Added option to Duplicate Checker to select all files except the highest resolution. ([#4286](https://github.com/stashapp/stash/pull/4286))

### 🎨 Improvements
* Show Performer image in Performer select list. ([#4227](https://github.com/stashapp/stash/pull/4227))
* Match Performers by alias during scraping and tagging if no Performer is found with the exact name (only if a single performer is found with the alias). ([#4182](https://github.com/stashapp/stash/pull/4182))
* Show Performer disambiguation and add stash-box links to Studio in tagger results. ([#4180](https://github.com/stashapp/stash/pull/4180))
* Show endpoints with stash ids. ([#4216](https://github.com/stashapp/stash/pull/4216))
* Load card thumbnails and similar images lazily when a component comes into view, instead of all at once. ([#4228](https://github.com/stashapp/stash/pull/4228))
* Made the random sorting more random. ([#4246](https://github.com/stashapp/stash/pull/4246))
* Added more VR projection modes. ([#3799](https://github.com/stashapp/stash/pull/3799))
* Improved the filtering behaviour when selecting a folder. ([#4277](https://github.com/stashapp/stash/pull/4277))
* Added support for setting plugins path from the UI. ([#4382](https://github.com/stashapp/stash/pull/4382))

### 🐛 Bug fixes
* Fixed submitting to stash-box not working after switching to another scene using the queue. ([#4354](https://github.com/stashapp/stash/pull/4354))
* Fixed UI crash when clearing a value from a URL or alias list. ([#4344](https://github.com/stashapp/stash/pull/4344))
* Fixed panic when exporting galleries. ([#4311](https://github.com/stashapp/stash/pull/4311))
* Fixed error when setting performer height with decimals. ([#4283](https://github.com/stashapp/stash/pull/4283))
* Fixed Performer stash ids being overwritten instead of merged when saving a Performer in the Performer tagger. ([#4215](https://github.com/stashapp/stash/pull/4215))
* Fixed organized being set to false if `Mark as Organized on save` is false when saving a scene in the tagger. ([#4213](https://github.com/stashapp/stash/pull/4213))
* Fixed URLs not populating correctly when scraping galleries. ([#4206](https://github.com/stashapp/stash/pull/4206))
* Fixed not being able to click next/previous scene if the applicable scenes had not been loaded by the queue. ([#4325](https://github.com/stashapp/stash/pull/4325))
* Fixed confirmation dialog not appearing on some screens when using the delete keyboard shortcut. ([#4387](https://github.com/stashapp/stash/pull/4387))
* Fixed desktop notifications not appearing on macos. ([#4153](https://github.com/stashapp/stash/pull/4153))
* Fixed video sometimes pausing when clicking on the scene scrubber. ([#4295](https://github.com/stashapp/stash/pull/4295))
* Fixed file paths not being shown when deleting image clips. ([#4323](https://github.com/stashapp/stash/pull/4323))
* Fixed image o-counts not being included in stats page. ([#4386](https://github.com/stashapp/stash/pull/4323))

<hr /><em>This discussion was created from the release <a href='https://github.com/stashapp/stash/releases/tag/v0.24.0'>v0.24.0</a>.</em>

**Comment by Maista6969** [comment]
Appending this here because it keeps coming up in Discord

The following query resolvers are being removed:
* `listPerformerScrapers` - Use `listScrapers(types: [PERFORMER])`
* `listSceneScrapers` - Use `listScrapers(types: [SCENE])`
* `listGalleryScrapers` - Use `listScrapers(types: [GALLERY])`
* `listMovieScrapers` - Use `listScrapers(types: [MOVIE])`
* `scrapePerformerList` - use `scrapeSinglePerformer`
* `scrapePerformer` - use `scrapeSinglePerformer`
* `scrapeScene` - use `scrapeSingleScene`
* `scrapeGallery` - use `scrapeSingleGallery`
* `scrapeFreeonesPerformerList` - use `scrapeSinglePerformer` with `scraper_id` = `builtin_freeones`

The following fields are being removed from `ConfigGeneralInput` and `ConfigGeneralResult`:
* `trustedProxies` - no longer supported
* `scraperUserAgent` - use field in `ConfigScraping[Input/Result]` instead
* `scraperCDPPath` - use field in `ConfigScraping[Input/Result]` instead
* `scraperCertCheck` - use field in `ConfigScraping[Input/Result]` instead

The following fields are being removed from `ConfigInterfaceInput` and `ConfigInterfaceResult`:
* `slideshowDelay` - Use `imageLightbox.slideshowDelay`
* `disabledDropdownCreate` - Use `disableDropdownCreate`

The following filter criteria are being removed from the applicable `FilterType` inputs:
* `height` - Use `height_cm` instead
* `stash_id` - Use `stash_id_endpoint` instead
* `rating` - Use 1-100 range with `rating100`

`SceneMarkerFilterType.tag_id` is being removed in favour of `tags`.

The `rating` field is being removed from all applicable types, in favour of the `rating100` field.

The `checksum`, `path` and `file_mod_time` fields are being removed `Scene`, `Gallery` and `Image` types. These can be accessed via the applicable `files` field (or `visual_files` in `image`). `oshash` and `phash` are also removed from `Scene`.

`checksum` is also removed from `Movie`, `Performer` and `Studio` types.

The following fields are being removed from `Performer` and applicable inputs (in addition to the changes above):
* `height` - use `height_cm` instead
* `aliases` - use `alias_list` instead

`chapters_vtt` is being removed from `ScenePathsType`.

`Gallery.images` is being removed. This can be accessed via `findImages` instead.

`file` is being removed from `Scene` and `Image` in favour of `Files`.

The `useFileMetadata` and `stripFileExtension` flags in `ScanMetadata[Input/Options]` are being removed. These were unused.

_Originally posted by @WithoutPants in https://github.com/stashapp/stash/issues/4064#issuecomment-1695010962_
